cmd/src/setup.rs

```diff
@@ -136,3 +136,21 @@ pub fn overwrite_config_with_cmd_args(config: &mut TiKvConfig, matches: &ArgMatc
         config.metric.address = metrics_addr.to_owned()
     }
 }
+
+#[allow(dead_code)]
+pub fn validate_config(config: &mut TiKvConfig, persist: bool) {
+    if let Err(e) = check_critical_config(config) {
+        fatal!("critical config check failed: {}", e);
+    }
+
+    if persist {
```

Comment:
if cannot extract the persistence logic out of `validate_config`, maybe should rename `validate_config`? eg, `check_and_persist_critical_config`?